### Data Cleaning

In [1]:
# import libraries
import pandas as pd
import numpy as np

import warnings
warnings.filterwarnings("ignore")

In [2]:
# Load the dataset
train = pd.read_csv("../data/raw/Train.csv")
test = pd.read_csv("../data/raw/Test.csv")

#### 1. Data Types

In [ ]:
train.dtypes.value_counts()

int64      121
float64     24
str          1
Name: count, dtype: int64

#### 2. Validate Id and Target

In [4]:
train["ID"].head()

0    ID_TR_NEW_XVGKFMLNRJ
1    ID_TR_NEW_GP8KNSWVP6
2    ID_TR_NEW_87X3957MVS
3    ID_TR_NEW_T4JMRPKHS3
4    ID_TR_NEW_2CTUQQ8KLU
Name: ID, dtype: str

In [5]:
print()
train["label"].value_counts().sort_index()

label
0    1086
1     735
Name: count, dtype: int64

In [6]:
print()
train["ID"].is_unique

True

The ID column contains unique values for every record, with no duplicate IDs found, indicating that each observation has a distinct identifier.

The dataset does not contain duplicate records or missing values, which indicates that the data is complete and does not require imputation or duplicate removal. 

The target variable (label) contains two classes: 0 with 1,086 samples and 1 with 735 samples. Although the classes are not perfectly balanced, the difference is moderate, suggesting that the dataset has a reasonably balanced distribution suitable for classification tasks.

Overall, the dataset is clean and ready for further preprocessing. The main preparation step is to remove the ID column before model training and apply appropriate feature preprocessing, such as scaling or normalization, if required by the selected machine learning algorithm.

In [9]:
import os
train.to_csv("../data/processed/cleaned_train.csv", index=False)